sdk/include/opentelemetry/sdk/common/attribute_utils.h

```diff
@@ -162,8 +216,36 @@ class AttributeMap : public std::unordered_map<std::string, OwnedAttributeValue>
     (*this)[std::string(key)] = nostd::visit(converter_, value);
   }
 
+  bool EqualTo(const opentelemetry::common::KeyValueIterable &attributes) const noexcept
+  {
+    if (attributes.size() != this->size())
+    {
+      return false;
+    }
+
+    const bool is_equal = attributes.ForEachKeyValue(
+        [this](nostd::string_view key,
+               const opentelemetry::common::AttributeValue &value) noexcept {
+          // Perform a linear search to find the key assuming the map is small
+          // This avoids temporary string creation from this->find(std::string(key))
+          for (const auto &kv : *this)
```

Comment:
Do you think we can use `this->find(kv.first)` here for better performance?

Reply:
It seems we have to start the search from the `KeyValueIterable::ForEachKeyValue` method and it provides a `nostd::string_view` key while the key type of the map is `std::string`.  So we either have to iterate directly like this or create a temp string every time  as in `auto iter = this->find(std::string(key))`. I chose the first option since it I've heard it could be faster than find for small maps :)  and I didn't feel like string allocation (and its potential to throw) was ideal.  I could profile both options if it seems worth it. 

C++20 gives the ["Heterogeneous comparison lookup" version of the `std::unordered_map::find` method](https://en.cppreference.com/w/cpp/container/unordered_map/find#Version_3) of unordered_map which is really what we need here.  Here's a neat example https://godbolt.org/z/Yqsx4rsY1

Reply:
Thanks for the details. Looks ok to me as is.